### DISCUSSION TITLE
Is there a way to convert the models in prisma to raw sql?

### DISCUSSION THREAD

**Discussion by SakoMe** [prisma/prisma#16812] [Q&A] [answered]
I'm using primsa as the ORM. Just a quick question:

Is there a way to see what the raw sql would look like without manually writing it? 

Basically taking this:

```
model User {
  id      Int      @id @default(autoincrement())
  Profile Profile?
}

model Profile {
  id   Int  @id @default(autoincrement())
  user Int  @unique
  User User @relation(fields: [user], references: [id])
}
```
and converting it to this:

```
CREATE TABLE "User" (
    id SERIAL PRIMARY KEY
);
CREATE TABLE "Post" (
    id SERIAL PRIMARY KEY,
    "author" integer NOT NULL,
    FOREIGN KEY ("author") REFERENCES "User"(id)
);
```

Is there a a command to run or a tool that exists that can take a schema written in primsa and convert it to raw SQL?

**Comment by janpio** [comment, marked as answer]
`npx prisma migrate dev --create-only` or a variant of `migrate diff`.

**Reply by SakoMe** [reply]
Hi jampio. I meant like output a file or something like that.

**Reply by janpio** [reply]
That is exactly what both `migrate dev --create-only` and `migrate diff` with some of the available parameters will do.

https://www.prisma.io/docs/reference/api-reference/command-reference#migrate-diff
`migrate diff --from-schema-datamodel prisma/schema.prisma --to-empty --script` probably

**Reply by 0xlakshan** [reply]
Just a heads up for anyone looking for an answer the following command helped me convert my schema.prisma file to a SQL file without needing an active database running. 
``npx prisma migrate diff --from-empty --to-schema-datamodel ./schema.prisma --script > output.sql`` worked for me

**Comment by zloom** [comment]
Faced same question while was creating integration tests, npx tools hard to integrate to tests pipelines ended up creating following tool https://github.com/zloom/prisma-sql-gen

**Comment by ludralph** [comment]
Hi there,

To keep our discussions organized and focused on the most relevant topics, we’re reviewing and tidying up our backlog. As part of this process, we’re closing discussions that have already been marked as answered but remain open.

If this discussion still requires further input or clarification, feel free to reopen it or start a new one with updated details. Your contributions are invaluable to the community, and we’re here to help!

For more details about our priorities and vision for the future of Prisma ORM, check out our latest blog post: https://www.prisma.io/blog/prisma-orm-manifesto.

Thank you for your understanding and ongoing support of the Prisma community!
